### PR TITLE
Fix DSL matching for Aggregation and GroupId nodes

### PR DESCRIPTION
**Pull request by rschlussel**
Use symbol aliases rather than symbols when looking for a match.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java [hunk 1]
```diff
@@ -82,26 +82,35 @@ public MatchResult detailMatches(PlanNode node, Session session, Metadata metada
...
 
-        for (T symbol : expected) {
-            if (!actual.contains(symbol)) {
+        int i = 0;
+        for (String alias : expectedAliases) {
+            SymbolReference reference = symbolAliases.get(alias);
+            boolean found = false;
+            while (i < actualSymbols.size()) {
+                Symbol outputSymbol = actualSymbols.get(i++);
+                if (reference.equals(outputSymbol.toSymbolReference())) {
```
Comment (ebd2) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java]: Why not convert to a Collection of actual symbols at the top of the function and use the old comparison logic?

Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java]: good point

Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java]: done

File: presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java [hunk 2]
```diff
@@ -82,26 +83,26 @@ public MatchResult detailMatches(PlanNode node, Session session, Metadata metada
...
+    static boolean matches(Collection<String> expectedAliases, Collection<Symbol> actualSymbols, SymbolAliases symbolAliases)
     {
-        if (expected.size() != actual.size()) {
+        if (expectedAliases.size() != actualSymbols.size()) {
             return false;
         }
 
-        for (T symbol : expected) {
-            if (!actual.contains(symbol)) {
+        List<Symbol> expectedSymbols = expectedAliases.stream().map(alias -> new Symbol(symbolAliases.get(alias).getName())).collect(toImmutableList());
```
Comment (ebd2) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java]: Formatting: one per line
```
stuff.stream()
    .map()
    .collect()
```

Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java]: fixed

**Pull request merged**